src/test/java/com/wavefront/sdk/common/metrics/WavefrontSdkMetricsRegistryTest.java

```diff
@@ -42,4 +42,25 @@ public void testCounter() {
     assertEquals(5, registry.newCounter("counter").count());
     assertEquals(0, registry.newCounter("counter2").count());
   }
+
+  @Test
+  public void testDeltaCounter() {
+    WavefrontSdkMetricsRegistry registry = new WavefrontSdkMetricsRegistry.Builder(null).
+        reportingIntervalSeconds(Integer.MAX_VALUE).
+        build();
+    WavefrontSdkDeltaCounter deltaCounter = registry.newDeltaCounter("deltaCounter");
+    assertEquals(0, deltaCounter.count());
+    deltaCounter.inc();
+    assertEquals(1, deltaCounter.count());
+    deltaCounter.inc(2);
+    assertEquals(3, deltaCounter.count());
+    deltaCounter.dec();
+    assertEquals(2, deltaCounter.count());
+    deltaCounter.dec(deltaCounter.count());
+    assertEquals(0, deltaCounter.count());
+
+    //Delta Counters decrements counters each time after data is sent. New counters with same name will have 0 count.
+    assertEquals(0, registry.newDeltaCounter("deltaCounter").count());
+    assertEquals(0, registry.newDeltaCounter("deltaCounter2").count());
+  }
```

Comment:
maybe also add a test to check that the value of the delta is reset after it is sent.